bottlerocket-os/bottlerocket
# containerd: share duplicate configs

**Issue number:**

Fixes https://github.com/bottlerocket-os/bottlerocket/pull/1511#discussion_r618823863

**Description of changes:**

```
containerd: share duplicate configs
    
Rather than naming the config file after the variant, use a more generic name,
and have each variant pick which generic config applies.  We only have two
different configs, so this immediately saves some duplication.
```

```
Add migration for containerd config template
```

**Testing done:**

For each of aws-k8s-1.19, aws-ecs-1, and aws-dev:
* Build and start a 1.0.8 image without these changes
* Confirm the original filenames in /usr/share/templates and that /etc/containerd/config.toml is correct
* Upgrade to 1.1.0 with these changes and migration
* Confirm the new (fewer) filenames in /usr/share/templates and that /etc/containerd/config.toml is still correct
* Confirm basic health of system (running pods/tasks/containers, API, services, etc.)
* Downgrade to 1.0.8, confirm old files again
* Confirm health again

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.
